nifi-nar-bundles/nifi-standard-services/nifi-hbase-client-service-api/src/main/java/org/apache/nifi/hbase/HBaseClientService.java

```diff
@@ -150,6 +194,40 @@
      */
     void scan(String tableName, byte[] startRow, byte[] endRow, Collection<Column> columns, ResultHandler handler) throws IOException;
 
+    /**
+     * Scans the given table for the given rowId and passes the result to the handler.
+     *
+     * @param tableName the name of an HBase table to scan
+     * @param startRow the row identifier to start scanning at
+     * @param endRow the row identifier to end scanning at
+     * @param columns optional columns to return, if not specified all columns are returned
+     * @param visibilityLabels optional list of visibility labels that the user should be able to see when communicating with HBase
+     * @param handler a handler to process rows of the result
+     * @throws IOException thrown when there are communication errors with HBase
+     */
+    void scan(String tableName, byte[] startRow, byte[] endRow, Collection<Column> columns, List<String> visibilityLabels, ResultHandler handler) throws IOException;
+
+    /**
+     * Get all of the labels in HBase.
+     *
+     * @return a List of all of the labels.
+     */
+    List<String> getLabels();
+
+    /**
+     * Get all of the labels a given user can see.
+     * @param user the user to lookup.
+     * @return a List of all of the labels a user is allowed to see.
+     */
+    List<String> getLabelsForUser(String user);
+
+    /**
+     * Get all of the labels the current user (NiFi process user or Kerberos keytab principle) can see.
+     *
+     * @return a List of all of the labels the current can see.
```

Comment:
current *user*?

Reply:
I have to give this some more thought, but HBase visibility labels work with or without a Kerberos configuration. So without it, the HBase client does simple auth and sends over the user running NiFi as the user.